File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0], result = 0;
        for(int i = 0; i < prices.size(); i++){
            minPrice = min(minPrice, prices[i]);

            result = max(result, prices[i] - minPrice);
        }

        return result;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        List<Integer> list=new ArrayList<>();
        for(int x:prices){
            list.add(x);
        }
        int maxi=Integer.MIN_VALUE;
        int mini=Integer.MAX_VALUE;
        for(int i=0; i<list.size(); i++){
            mini=Math.min(mini,list.get(i));
            maxi=Math.max(maxi,list.get(i)-mini);
            
        }
        return maxi;
        
    }
}